src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -148,33 +149,53 @@ void reconnect() throws TimeoutException, ExecutionException, InterruptedExcepti
         }
     }
 
-    @SuppressWarnings("PMD.NullAssignment")
+
     protected synchronized CompletableFuture<Boolean> connect() {
         if (connection != null) {
             return CompletableFuture.completedFuture(true);
         }
+        // For the initial connect, client connects with cleanSession=true and disconnects.
+        // This deletes any previous session information maintained by IoT Core.
+        // For subsequent connects, the client connects with cleanSession=false
+        if (initalConnect.get()) {
+            try {
+                establishConnection(true).get(getTimeout(), TimeUnit.MILLISECONDS);
```

Comment:
this is now a blocking call, so you might want to consider making `connect` more asynchronous so that we don't block threads for a long time.

Reply:
refactored to make connect asynchronous.